src/features/profile/components/UserProfile/RenameUserForm.tsx

```diff
@@ -72,6 +76,9 @@ export const RenameUserForm = ({ userId, currentName, isOpen, onClose }: IRename
    * @param event - The keyboard event
    */
   const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (isLoading) {
+      return; // Do nothing if isLoading is true
+    }
```

Comment:
This part is unnecessary because [ escape ] key down does not trigger a submit.
Multiple submissions by [ enter ] key down can be avoided within the processsSubmit, which you have already modified.